umbral/point.py

```diff
@@ -194,11 +196,12 @@ def __sub__(self, other):
         """
         Performs subtraction by adding the inverse of the `other` to the point.
         """
-        return (self + (~other))
+        return (self + (-other))
 
-    def __invert__(self) -> 'Point':
+    def __neg__(self) -> 'Point':
```

Comment:
I like this, but my only concern for this is that someone accidentally subs when they want to invert.

For example, you must do `(-foobar)` in this case to make it visually clear that you are doing an inversion.

Look at this line for example:
`a = b - -c`

This is a valid line, but it looks _weird._ The obvious answer is to do:
`a = b - (-c)`

It no longer looks _weird_, but I think it could be accidentally looked over. I like `__invert__` here because, while not mathematically correct, it looks far more interesting visually and I think would reduce the chances of a typo.

What do you think?

Reply:
Also, if you do `a = b-c`, though not proper, is still valid and makes readability more problematic.

Reply:
I think that choice would be far weirder. Python itself allows for expressions like `a = b - -c`

Reply:
Yeah, but just because python allows doesn't make it good. If we're going down that road, there's a lot of horrible naughty things that Python allows.

Reply:
We don't have to go down that road, true. But going back to the question, I really don't see what's the issue, or what do you want to protect users against. Using the minus sign to represent the opposite or additive inverse of an element is something strongly established, as you can see in papers, textbooks and programming languages (e.g., a quick reference: http://mathworld.wolfram.com/AdditiveInverse.html). 

Actually, from an abstract algebra point of view, in a group, _subtraction doesn't exist_. What does exist in a group is addition and the opposite, and subtraction is mere syntactic sugar on top of this. So if we accept that the sign for subtraction is the minus, then it doesn't make sense to use a different symbol to the actual operator underneath.

Reply:
I'm not necessarily trying to protect the users from anything, but I'm trying to protect the readability of the code. It would be pretty easy to mistake a negation for a subtraction.

Does this mean you're willing to get rid of subtraction and simply add a negation?

Reply:
Ah. Removing subtraction while having negation would actually be neat.

Although, I'm not against the original `~`. But if negation is technically more precise, so be it

Reply:
I didn't mean to get rid of subtraction. As many other times, I tend to push towards following long established conventions. In this case, to use the same sign for the negation and subtraction operators. There's no risk of mistaking anything, they are different types of operators (unary vs binary), with different precedence orders. I simply don't see any value of using two different operators.

Reply:
Alright, I'm sold.